cpp/src/arrow/array/array_list_test.cc

```diff
@@ -197,6 +197,37 @@ class TestListArray : public ::testing::Test {
     EXPECT_FALSE(left->Slice(offset)->Equals(right->Slice(offset)));
   }
 
+  void TestFromArraysWithNullBitMap() {
+    std::shared_ptr<Array> offsets_w_nulls, offsets_wo_nulls, values;
+
+    std::vector<offset_type> offsets = {0, 1, 1, 3, 4};
+
+    std::vector<bool> offsets_w_nulls_flags = {true, false, true, true, true};
+    std::vector<bool> offsets_wo_nulls_flags = {true, true, true, true, true};
+
+    std::vector<int32_t> null_bitmap_values = {1, 0, 1, 1};
+    std::shared_ptr<Buffer> null_bitmap = Buffer::Wrap(null_bitmap_values);
+
+    ArrayFromVector<OffsetType, offset_type>(offsets_w_nulls_flags, offsets,
+                                             &offsets_w_nulls);
+    ArrayFromVector<OffsetType, offset_type>(offsets_wo_nulls_flags, offsets,
+                                             &offsets_wo_nulls);
+
+    auto type = std::make_shared<T>(int32());
+    auto expected = std::dynamic_pointer_cast<ArrayType>(
+        ArrayFromJSON(type, "[[0], null, [0, null], [0]]"));
+
+    values = expected->values();
+
+    ASSERT_OK(ArrayType::FromArrays(*offsets_w_nulls, *values, pool_));
+    ASSERT_OK(ArrayType::FromArrays(*offsets_wo_nulls, *values, pool_));
```

Comment:
Ok, but can we also test the array that was just created? See `TestFromArrays` below for use of `ValidateFull` and `AssertArraysEqual`.